core/tracing/hooks.go

```diff
@@ -217,6 +217,16 @@ type Hooks struct {
 	OnBlockHashRead BlockHashReadHook
 }
 
+// HooksState returns if any of the state events are being hooked
+func (h *Hooks) HooksState() bool {
```

Comment:
Maybe `HasStateHooks` ? HooksState is correct, but sounds a bit weird (double meaning of hooks and Hooks)

Reply:
Yeap, that definitely is easier to understand.